### DISCUSSION TITLE
bug - sqlalchemy.exc.ArgumentError: SQL expression for WHERE/HAVING role expected, got

### DISCUSSION THREAD

**Discussion by marti1125** [sqlalchemy/sqlalchemy#8275] [Usage Questions]
I try to map this table

metadata_obj = MetaData()

ParamConfig = Table("param_config", metadata_obj,
                    Column("param_id"),
                    Column("param_name"))

query = ParamConfig.select([ParamConfig]).where(ParamConfig.c.param_id == 7)
query.compile(compile_kwargs={"literal_binds": True})

sqlalchemy.exc.ArgumentError: SQL expression for WHERE/HAVING role expected, got

**Comment by CaselIT** [comment]
Hi,

You have to specify colum types.
The default null type has no implementation

**Reply by marti1125** [reply]
I still have the same error
Table("param_config", metadata_obj,
                    Column("fi_param_id", Integer),
                    Column("fc_param_name", String))

**Reply by CaselIT** [reply]
Can you provide a runnable example? You can look at https://github.com/sqlalchemy/sqlalchemy/issues?q=label%3A%22great+mcve%22+is%3Aclosed for examoles of runnable mcve

**Reply by marti1125** [reply]
I find the problem I think the method select from ParamConfig = Table I got the an error 
but when i try with 
from sqlalchemy import select
 select([ParamConfig]).where(ParamConfig.c.fi_param_id == 7)
 it works
